liquibase-core/src/main/java/liquibase/ThreadLocalScopeManager.java

```diff
@@ -0,0 +1,42 @@
+package liquibase;
+
+/**
+ * An alternative to {@link SingletonScopeManager} which manages a separate Scope per thread.<br><br>
+ * Integrations that would prefer to use this scope manager can call <pre>Scope.setScopeManager(new ThreadLocalScopeManager())</pre>.
+ * <br><br>
+ * The value of Scope.getCurrentScope() at the time of the ThreadLocalScopeManger's creation will be the basis of all scopes created after setScopeManager() is changed,
+ * so you will generally want to setScopeManager as soon as possible.
+ */
+@SuppressWarnings("java:S5164")
+public class ThreadLocalScopeManager extends ScopeManager {
+
+    private final Scope rootScope;
+    private final ThreadLocal<Scope> threadLocalScopes = new ThreadLocal<>();
+
+    ThreadLocalScopeManager() {
```

Comment:
- [ ] Make ThreadLocalScopeManager constructor public, it is intended for public use.

Reply:
Good point, I fixed that up. Thanks